blockchain/blockchain.go

```diff
@@ -252,6 +252,13 @@ func NewBlockChain(db database.DBManager, cacheConfig *CacheConfig, chainConfig
 		prefetchTxCh:       make(chan prefetchTx, MaxPrefetchTxs),
 	}
 
+	// set hardForkBlockNumberConfig which will be used as a global variable
+	if err := fork.SetHardForkBlockNumberConfig(&params.ChainConfig{
```

Comment:
Do you have any reason why you create new ChainConfig? instead of passing the `bc.chainConfig`?

Reply:
Also, why this method is called in `NewBlockChain`? How about setting hardfork block number right after chain config creation?

Reply:
> Do you have any reason why you create new ChainConfig? instead of passing the bc.chainConfig?

I wanted to pass hard fork information only. However, it would be more simple if bc.chainConfig is passed. thanks!

> why this method is called in NewBlockChain? How about setting hardfork block number right after chain config creation?

I called this method in here because i want to call it right after when `Blockchain` struct is constructed.